### ISSUE TITLE
Not premium accounts issues with delegate methods

### ISSUE THREAD

**Issue by Ayc0** [spotify/ios-sdk#886]
## Description
The delegates methods aren't called if the account isn't a premium one

### Versions
iOS SDK version
Spotify app version 25

### Severity
Major

## Steps to Reproduce
Just login with a non premium account

### Actual behaviour
Neither `audioStreamingDidLogin:(SPTAudioStreamingController *)audioStreaming` nor `audioStreaming:(NSError *)didEncounterError audioStreaming:(SPTAudioStreamingController *)audioStreaming` are called after an authentication with a non premium account

### Expected behaviour
`audioStreaming:(NSError *)didEncounterError audioStreaming:(SPTAudioStreamingController *)audioStreaming` should be called

**Comment by hughrawlinson** [comment]
Hi @Ayc0 - as noted in [our documentation](https://developer.spotify.com/technologies/spotify-ios-sdk/), streaming is not available for non-premium accounts.

**Comment by Ayc0** [comment]
Indeed, but if you integrate this in an app and if a user without a premium account tries to connect to the app, you have no other way to detect this, so `audioStreaming:(NSError *)didEncounterError audioStreaming:(SPTAudioStreamingController *)audioStreaming` should be triggered

**Comment by Ayc0** [comment]
and if we could detect when a user doesn't have a premium account, we would be able to encourage them to buy it

**Comment by hughrawlinson** [comment]
Hi! You can detect this using the `product` field of the `SPTUser` object of the current user. Documentation [here](https://spotify.github.io/ios-sdk/Classes/SPTUser.html). We agree, it'd be better to throw an error though, so we'll keep this ticket open.
